internal/ingress/metric/collectors/socket.go

```diff
@@ -415,6 +419,12 @@ func (sc *SocketCollector) RemoveMetrics(ingresses []string, registry prometheus
 					klog.V(2).InfoS("metric not removed", "name", metricName, "ingress", ingKey, "labels", labels)
 				}
 			}
+
+			if c, ok := metric.(*prometheus.CounterVec); ok {
+				if removed := c.Delete(labels); !removed {
+					klog.V(2).InfoS("metric not removed", "name", metricName, "ingress", ingKey, "labels", labels)
+				}
+			}
```

Comment:
additionally during unit-testing, i also discovered that `nginx_ingress_controller_requests` would not get removed/deleted in the unit-test (by the call of `sc.RemoveMetrics(...)`) due to `prometheus.CounterVec` was missing from here.